src/Sentry/Extensibility/ISentryTransactionProcessor.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace Sentry.Extensibility
+{
+    /// <summary>
+    /// Process a <see cref="Transaction"/> during the prepare phase.
+    /// </summary>
+    public interface ISentryTransactionProcessor
+    {
+        /// <summary>
+        /// Process the <see cref="Transaction"/>
+        /// </summary>
+        /// <param name="transaction">The Transaction to process</param>
+        void Process(Transaction transaction);
```

Comment:
Are we sure we don't want to use a nullable return value, like we do for event processors?  With this approach, you can only mutate the existing transaction - you can't create a new one.